flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/sinks/PrintTableSinkFactory.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.sinks;
+
+import org.apache.flink.annotation.Experimental;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.streaming.api.functions.sink.SinkFunction;
+import org.apache.flink.table.connector.ChangelogMode;
+import org.apache.flink.table.connector.sink.DynamicTableSink;
+import org.apache.flink.table.connector.sink.SinkFunctionProvider;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.factories.DynamicTableSinkFactory;
+import org.apache.flink.table.types.DataType;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Print table sink factory.
+ */
+@Experimental
+public class PrintTableSinkFactory implements DynamicTableSinkFactory {
+
+	public static final String IDENTIFIER = "print";
+
+	@Override
+	public String factoryIdentifier() {
+		return IDENTIFIER;
+	}
+
+	@Override
+	public Set<ConfigOption<?>> requiredOptions() {
+		return new HashSet<>();
+	}
+
+	@Override
+	public Set<ConfigOption<?>> optionalOptions() {
+		return new HashSet<>();
+	}
+
+	@Override
+	public DynamicTableSink createDynamicTableSink(Context context) {
+		return new PrintSink(context.getCatalogTable().getSchema().toRowDataType());
+	}
+
+	static class PrintSink implements DynamicTableSink {
+
+		private final DataType type;
+
+		private PrintSink(DataType type) {
+			this.type = type;
+		}
+
+		@Override
+		public ChangelogMode getChangelogMode(ChangelogMode requestedMode) {
+			return requestedMode;
+		}
+
+		@Override
+		public SinkRuntimeProvider getSinkRuntimeProvider(DynamicTableSink.Context context) {
+			DataStructureConverter converter = context.createDataStructureConverter(type);
+			return SinkFunctionProvider.of(new SinkFunction<RowData>() {
```

Comment:
Could you extract this SinkFunction as a static class?

Reply:
I'd like to use `PrintSinkFunction`.

Reply:
`PrintSinkFunction` is not suitable for our case, because it will just simply print to `toString`.

Reply:
Yeah, I want to extend or invoke it in our sink function.